setup.py

```diff
@@ -74,6 +74,11 @@ def find_sources():
     platform_specific['define_macros'] = [('WIN32', 1)]
     jni_md_platform = 'win32'
 
+elif sys.platform == 'cygwin':
```

Comment:
I will defer on the PyUnicode,  I was struggling to find a set of patterns that worked with the compiler.  It seems like that was a left over.  I will try to get Appveyor working, but I don't know much about it.